### PR TITLE
ENG-17144 Node SDK should be updated to use V2 of AgentStationGroup endpoints; It should also remove Drop from RetryPolicy

### PR DESCRIPTION
**Pull request by yangAtSpoken**

### CODE REVIEW

**Review by rbshaik — APPROVED** [review]
Looks good to me.

File: NodeJS/__integration__/RestClientSubscription.it.ts [hunk 1]
```diff
@@ -40,19 +41,19 @@ describe('RestClient Subscription Integration Test', () => {
     expect(subscriptions).toEqual({})
   })
 
-  test('createDataSubscription, updateDataSubscription, and deleteDataSubscription should work as expected.', async () => {
+  test.only('createDataSubscription, updateDataSubscription, and deleteDataSubscription should work as expected.', async () => {
```
Comment (eric3chang) [inline NodeJS/__integration__/RestClientSubscription.it.ts:44]: are the other tests not working, so we have to use test.only?

**Review by eric3chang — DISMISSED** [review]
found a test that had only added to it

File: NodeJS/__integration__/RestClientSubscription.it.ts (hunk 1 above)
Comment (yangAtSpoken) [reply NodeJS/__integration__/RestClientSubscription.it.ts:44]: These integration tests depends on the environment. They work when they are properly set up.  "only" is a convenient way to run just the test you are interested at the moment.

File: NodeJS/__integration__/RestClientAgent.it.ts [hunk 2]
```diff
@@ -23,21 +30,23 @@ describe('RestClient.ts integration test', () => {
...
-      clientIdWithoutAgentStationGroups
+      subAccountAppIdWithoutAgentStationGroups
     )
     expect(id).toEqual(STATION_GROUP_ID_NOT_EXISTS)
   })
-  test('getAgentStationGroupId should return id if no station groups exist', async () => {
-    const clientIdNotExists = '11111'
-    const id = await restClient.getAgentStationGroupId(clientIdNotExists)
-    expect(id).toEqual(-403)
+  test('getAgentStationGroupId should return -404 if no station groups exist', async () => {
```
Comment (joe-s-avaya) [inline NodeJS/__integration__/RestClientAgent.it.ts:46]: 👍

File: NodeJS/src/RestClient.ts [hunk 3]
```diff
@@ -26,10 +26,16 @@ import { SkillCreateRequest, Subscription } from './models'
...
 const axios = require('axios').default
+
 const axiosCookieJarSupport = require('@3846masa/axios-cookiejar-support')
   .default
 const logger = log4js.getLogger('RestClient')
 
+axios.interceptors.request.use((request: any) => {
+  logger.debug('Starting Request', JSON.stringify(request, null, 2))
+  return request
+})
```
Comment (joe-s-avaya) [inline NodeJS/src/RestClient.ts:37]: is this interrupting ALL api calls? or am I misunderstanding?

Comment (yangAtSpoken) [reply NodeJS/src/RestClient.ts:37]: This will log all requests if level is at debug.  Do not think it interrupt a call other than logging the request.

Comment (joe-s-avaya) [reply NodeJS/src/RestClient.ts:37]: cool, I was misunderstanding how that worked, thanks for clarifying!

**Review by joe-s-avaya — APPROVED** [review]

**Pull request merged**